compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/binary/KSTypeReferenceDescriptorImpl.kt

```diff
@@ -25,21 +25,26 @@ import com.google.devtools.ksp.symbol.impl.memoized
 import org.jetbrains.kotlin.builtins.isSuspendFunctionTypeOrSubtype
 import org.jetbrains.kotlin.types.KotlinType
 
-class KSTypeReferenceDescriptorImpl private constructor(val kotlinType: KotlinType, override val origin: Origin) :
-    KSTypeReference {
-    companion object : KSObjectCache<IdKey<Pair<KotlinType, Origin>>, KSTypeReferenceDescriptorImpl>() {
-        fun getCached(kotlinType: KotlinType, origin: Origin) =
-            cache.getOrPut(IdKey(Pair(kotlinType, origin))) { KSTypeReferenceDescriptorImpl(kotlinType, origin) }
+class KSTypeReferenceDescriptorImpl private constructor(
+    val kotlinType: KotlinType,
+    override val origin: Origin,
+    override val parent: KSNode?
+) : KSTypeReference {
+    companion object : KSObjectCache<IdKey<Triple<KotlinType, Origin, KSNode?>>, KSTypeReferenceDescriptorImpl>() {
+        fun getCached(kotlinType: KotlinType, origin: Origin, parent: KSNode?) = cache
+            .getOrPut(IdKey(Triple(kotlinType, origin, parent))) {
```

Comment:
Found a critical issue in the original implementation: Triple/Pair shouldn't be used to construct IdKey; They'll always compare to false and never cached. What's desired is `IdKey2` / `IdKey3`.

`TypeReferenceDescriptorImpl` and `TypeArgumentDescriptorImpl` was bad :(